container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/backend_creator/backend_creator.go

```diff
@@ -73,10 +78,35 @@ func getLocalDockerKurtosisBackend(
 	optionalApiContainerModeArgs *APIContainerModeArgs,
 ) (backend_interface.KurtosisBackend, error) {
 	dockerClientOpts := []client.Opt{
-		client.FromEnv,
 		client.WithAPIVersionNegotiation(),
 	}
 
+	// If the DOCKER_HOST env variable is set, use it. Otherwise, try to locate the daemon socket. Otherwise, fall back
+	// to env variables as we were doing before
+	userHomeDir, err := os.UserHomeDir()
```

Comment:
i wonder if we can look t his up after `client.EnvOverrideHost` and `systemDaemonSocket` have failed to yield a `dockerClientOpts`; as we don't use this otherwise and the debug line could be mis-leading

but the current way is definitely more readable

Reply:
We don't throw if it returns an error, so technically it doesn't affect the code flow, and I find it more readable this way.